decred/decred/dcr/dcrdata.py

```diff
@@ -396,17 +395,18 @@ class DcrdataBlockchain:
     def __init__(self, db, params, datapath, skipConnect=False):
         """
         Args:
-            db (str | database.Bucket): the database bucket or a filepath.
-                If a filepath, a new database will be created.
+            db (database.Bucket | database.KeyValueDatabase | filepath):
+                the database or a filepath. If a filepath, a new database
+                will be created.
             params obj: blockchain network parameters.
             datapath str: a URI for a dcrdata server.
             skipConnect bool: skip initial connection to dcrdata.
         """
         # Allow string arguments for database.
         self.ownsDB = False
-        if isinstance(db, str):
+        if not isinstance(db, (database.Bucket, database.KeyValueDatabase)):
```

Comment:
The `sqlite3` library is able to create database files using filepath objects. Removing the `str` requirement allows us to avoid string conversions in calling code.

By the way, is the ability to pass an already created database object actually useful?

Reply:
The idea is that the caller can decide whether to the blockchain gets its own database instance, or whether it creates tables in e.g. an application database. If we're not using the feature, I'm okay with removing it. Feel free to follow up.